private/pkg/protosource/protosource.go

```diff
@@ -151,6 +151,8 @@ type OptionExtensionDescriptor interface {
 	// PresentExtensionNumbers returns field numbers for all options that
 	// have a set value on this descriptor.
 	PresentExtensionNumbers() []int32
+
+	PresentFields() map[protoreflect.FieldDescriptor]protoreflect.Value
```

Comment:
Needs doc comment. Here's a suggestion:
```suggestion
	// PresentFields returns a map of field descriptor to value for all
	// option fields present. This includes both normal option fields
	// and extension fields (aka custom options).
	PresentFields() map[protoreflect.FieldDescriptor]protoreflect.Value
```